File: String/LongestCommonPrefix.java
class Solution {
    public int longestCommonPrefix(int[] arr1, int[] arr2) {
        int ans=0;
        HashSet<String> set = new HashSet<>();
        for(int num:arr1){
            String s=Integer.toString(num);
            String temp="";

            for(char ch:s.toCharArray()){
                temp+=ch;
                set.add(temp);
            }
        }
        for(int num:arr2){
            String s=Integer.toString(num);
            String temp="";

            for(char ch:s.toCharArray()){
                temp+=ch;
                if(set.contains(temp)){
                    ans=Math.max(ans,temp.length());
                }
            }
        }
        return ans;
    }
}

File: Strings/LongestCommonPrefix.cpp
// Problem: Longest Common Prefix
// Platform: LeetCode
class Solution {
public:
    int longestCommonPrefix(vector<int>& arr1, vector<int>& arr2) {
        int ans=0;
        unordered_set<string> set;
        for(auto num:arr1){
            string s=to_string(num);
            string temp="";
            for(char ch:s){
                temp+=ch;
                set.insert(temp);
            }
        }
        for(auto num:arr2){
            string s=to_string(num);
            string temp="";
            for(char ch:s){
                temp+=ch;
                int n=temp.length();
                if(set.find(temp)!=set.end()){
                    ans=max(ans,n);
                }
            }
        }
        return ans;
    }
};
